### ISSUE TITLE
A11y: Toolbar items do not have the right accessibility settings

### ISSUE THREAD

**Issue by jantanis** [xamarin/Xamarin.Forms#14421]
### Description
To be WCAG compliant buttons should have the "Button" a11y label. Toolbar items do not have this on Android.

### Steps to Reproduce

1.  Launch sample app
2. Activate talk back
3. Select the toolbar items

### Expected Behavior
Every clickable item should have 
`
info.ClassName = "android.widget.Button";
`

### Actual Behavior
Toolbar items do not have this set. And are therefore not spoken as button. The "More option" (secondary) item does have this set. It's childs doesn't.

### Basic Information

- Version with issue: 5.0.0.2012
- Last known good version:
- Platform Target Frameworks: 
  - Android: Look at environment info
- Android Support Library / AndroidX Version: 
- NuGet Packages:
- Affected Devices:

### Environment
<details><summary>Show/Hide Visual Studio info</summary>

```
=== Visual Studio Enterprise 2019 for Mac ===

Version 8.10.6 (build 10)
Installation UUID: a186dc20-9f9a-4d08-bada-2607f2d634ad
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 6.18.0.23 (d16-6 / 088c73638)

	Package version: 612000140

=== Mono Framework MDK ===

Runtime:
	Mono 6.12.0.140 (2020-02/51d876a041e) (64-bit)
	Package version: 612000140

=== Roslyn (Language Service) ===

3.10.0-4.21269.26+029847714208ebe49668667c60ea5b0a294e0fcb

=== NuGet ===

Version: 5.9.0.7134

=== .NET Core SDK ===

SDK: /usr/local/share/dotnet/sdk/6.0.100-preview.3.21202.5/Sdks
SDK Versions:
	6.0.100-preview.3.21202.5
	5.0.302
	5.0.301
	5.0.203
	5.0.202
	5.0.201
	5.0.103
	5.0.102
	5.0.101
	5.0.100
	3.1.411
	3.1.410
	3.1.409
	3.1.408
	3.1.407
	3.1.406
	3.1.405
	3.1.404
	3.1.403
	3.1.402
	3.1.401
	3.1.302
	3.1.301
	3.1.300
	3.1.202
	3.1.200
	3.1.102
	3.1.101
	3.1.100
	3.0.101
	3.0.100
	2.2.100
	2.1.810
	2.1.809
	2.1.701
	2.1.700
	2.1.505
MSBuild SDKs: /Applications/Visual Studio.app/Contents/Resources/lib/monodevelop/bin/MSBuild/Current/bin/Sdks

=== .NET Core Runtime ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	6.0.0-preview.3.21201.4
	5.0.8
	5.0.7
	5.0.6
	5.0.5
	5.0.4
	5.0.3
	5.0.2
	5.0.1
	5.0.0
	3.1.17
	3.1.16
	3.1.15
	3.1.14
	3.1.13
	3.1.12
	3.1.11
	3.1.10
	3.1.9
	3.1.8
	3.1.7
	3.1.6
	3.1.5
	3.1.4
	3.1.2
	3.1.1
	3.1.0
	3.0.1
	3.0.0
	2.2.0
	2.1.23
	2.1.22
	2.1.21
	2.1.20
	2.1.19
	2.1.18
	2.1.17
	2.1.16
	2.1.15
	2.1.14
	2.1.13
	2.1.12
	2.1.11
	2.1.9

=== .NET Core 3.1 SDK ===

SDK: 3.1.411

=== Xamarin.Profiler ===

Version: 1.6.12.29
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 12.5.1 (18212)
Build 12E507

=== Xamarin Designer ===

Version: 16.10.0.119
Hash: 36a2d986f
Branch: remotes/origin/d16-10
Build date: 2021-06-02 19:41:34 UTC

=== Xamarin.Mac ===

Xamarin.Mac not installed. Can't find /Library/Frameworks/Xamarin.Mac.framework/Versions/Current/Version.

=== Xamarin.iOS ===

Version: 14.20.0.24 (Visual Studio Enterprise)
Hash: c4b89cddb
Branch: d16-10
Build date: 2021-06-15 22:03:01-0400

=== Xamarin.Android ===

Version: 11.3.0.4 (Visual Studio Enterprise)
Commit: xamarin-android/d16-10/ae14caf
Android SDK: /Users/jan/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		8.1 (API level 27)

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 31.0.2
SDK Build Tools Version: 30.0.3

Build Information: 
Mono: b4a3858
Java.Interop: xamarin/java.interop/d16-10@f39db25
ProGuard: Guardsquare/proguard/v7.0.1@912d149
SQLite: xamarin/sqlite/3.35.4@85460d3
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-10@c5732a0

=== Microsoft OpenJDK for Mobile ===

Java SDK: /Users/jan/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android SDK Manager ===

Version: 16.10.0.12
Hash: e240b8c
Branch: remotes/origin/d16-10
Build date: 2021-06-01 18:26:34 UTC

=== Android Device Manager ===

Version: 16.10.0.14
Hash: e340248
Branch: remotes/origin/d16-10
Build date: 2021-06-01 18:26:52 UTC

=== Build Information ===

Release ID: 810060010
Git revision: 263d2fc9398e8092c231fa5d310cd72205cd627c
Build date: 2021-07-08 14:37:29-04
Build branch: release-8.10

=== Operating System ===

Mac OS X 10.16.0
Darwin 20.3.0 Darwin Kernel Version 20.3.0
    Thu Jan 21 00:07:06 PST 2021
    root:xnu-7195.81.3~1/RELEASE_X86_64 x86_64

=== Enabled user installed extensions ===

DeepClean 1.2.5
NuGet Package Management Extensions 0.27
MFractor 4.4.8
Straight8's SpecFlow Integration 1.11.3.0
AapIdeExtension.Mac 1.1.81


```

</details>

### Reproduction Link
[A11y.Toolbar.zip](https://github.com/xamarin/Xamarin.Forms/files/6822115/A11y.Toolbar.zip)

### Workaround
So far I was able to get it working for primary toolbar items by setting a A11y delegate on ActionMenuView_ChildViewAdd that sets the classname. I was not (yet) able to do this for secondary items.

**Comment by jfversluis** [comment]
@rachelkang did you catch this one yet?

**Comment by rachelkang** [comment]
Hi, @jantanis - thank you for the feedback, and @jfversluis thanks for bringing this to my attention!

The behavior you are observing is indeed the expected behavior - the toolbar items don't get announced as a "button". Due to the way the controls are designed on Android, it is technically _not_ a button. As Android does not classify and therefore doesn not announce it as a button, we are consistent with that behavior in Xamarin.Forms. However, TalkBack does still announce "Double tap to activate", which makes the item accessible as a button.

That being said, you definitely have the right idea for the code that would be needed to achieve what you're looking to. I highly recommend checking out @PureWeen's https://github.com/PureWeen/A11YTools; the `SetControlType` and `SetIsClickable` methods should be of particular interest to you!

I'm not sure that we need to make _all_ clickable items need to be classified as buttons, as that is not needed for accessibility via TalkBack. However, if you have further information or believe I am misunderstanding something, please feel free to share!

**Comment by jfversluis** [comment]
Assuming this new information helped, let us know if there is still something we can do!
